File: algorithm/#1/twoSum.go
func twoSum(nums []int, target int) []int {
	hashmap := make(map[int]int)
	for i, v := range nums {
		temp := target - v
		if j, ok := hashmap[temp]; ok {
			return []int{i, j}
		} else {
			hashmap[v] = i
		}
	}
	return nil
}

File: algorithm/#1/twoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        for ((i, v) in nums.withIndex()) {
            val num = map[target - v]
            if (num != null){
                return intArrayOf(i, num)
            } else {
                map[v] = i
            }
        }
        return intArrayOf()
    }
}